### PR TITLE
 Auto retrieval of a LetsEncrypt cert, better ports

### PR DESCRIPTION
**Pull request by tfenster**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

**Comment by tfenster** [comment]
Hi @bmoore-msft, do I need to change anything here to get it merged?

File: 101-aci-dynamicsnav/azuredeploy.json [hunk 1]
```diff
@@ -99,6 +105,14 @@
                 { 
                   "name": "PublicDnsName",
                   "value": "[variables('publicdnsname')]"
+                },
+                {
+                  "name": "folders",
+                  "value": "c:\\run\\my=https://github.com/NAVDEMO/LetsEncrypt/archive/master.zip\\LetsEncrypt-master"
```
Comment (bmoore-msft) [inline 101-aci-dynamicsnav/azuredeploy.json]: this file needs to come from this repo

File: 101-aci-dynamicsnav/azuredeploy.json [hunk 2]
```diff
@@ -99,6 +105,14 @@
                 { 
                   "name": "PublicDnsName",
                   "value": "[variables('publicdnsname')]"
```
Comment (bmoore-msft) [inline 101-aci-dynamicsnav/azuredeploy.json]: this needs to come from the reference() function - as this is written will fail when the location parameter is used

File: 101-aci-dynamicsnav/azuredeploy.json (hunk 1 above)
Comment (tfenster) [reply 101-aci-dynamicsnav/azuredeploy.json]: @bmoore-msft Then I can't really test before the merge, right?

File: 101-aci-dynamicsnav/azuredeploy.json (hunk 2 above)
Comment (tfenster) [reply 101-aci-dynamicsnav/azuredeploy.json]: @bmoore-msft I think we've had that discussion here https://github.com/Azure/azure-quickstart-templates/pull/4616/files/a3380d5851b8caad0ce09faafe22412ed178b5c7 before or am I misunderstanding you? I need to have the fqdn when the container starts and don't know how to achieve that

Comment (bmoore-msft) [reply 101-aci-dynamicsnav/azuredeploy.json]: Alright - we need to find a solution for this... we're not going to have this pattern repeated in the repo.  I would recommend finding out where that re-entrant requirement is coming from and see how that can be addressed.

Comment (tfenster) [reply 101-aci-dynamicsnav/azuredeploy.json]: @bmoore-msft The only thing I can come up with would be the following:

1. Run a script at the very beginning of container startup which calls an Azure Function
2. That Azure Function tries to take the requesting IP (and I hope this is the public IP of the container)
3. I run something like nslookup to determine the FQDN of that IP

Kind of crude but I don't see an alternative. Do you see any obvious problems with that approach

**Comment by tfenster** [comment]
@bmoore-msft any feedback on this?

File: 101-aci-dynamicsnav/azuredeploy.json [hunk 3]
```diff
@@ -60,17 +66,27 @@
...
+      "defaultValue": "[resourceGroup().location]",	
+      "metadata": {	
+        "description": "Location for all resources."	
       }
     }
   },
   "variables": {
     "image": "[parameters('navRelease')]",
-    "publicdnsname": "[concat(parameters('dnsPrefix'),'.',resourceGroup().location,'.azurecontainer.io')]"
+    "publicdnsname": "[concat(parameters('dnsPrefix'),'.',resourceGroup().location,parameters('azurecontainerSuffix'))]"
```
Comment (bmoore-msft) [inline 101-aci-dynamicsnav/azuredeploy.json]: need to use parameters('location') here...

File: 101-aci-dynamicsnav/azuredeploy.json (hunk 1 above)
Comment (bmoore-msft) [reply 101-aci-dynamicsnav/azuredeploy.json]: See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#deployment-artifacts-nested-templates-scripts) for how to handle artifacts

**Comment by tfenster** [comment]
@bmoore-msft thanks a lot for your help! I finally understood how the staging works and now the ACI starts and works as expected. Do you need any more changes?

**Pull request merged**